server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -755,16 +755,18 @@ private Symbol resolveUnindexedSubscriptExpression(
                     );
                 }
                 DataType<?> currentType = baseType;
-                ColumnPolicy parentPolicy = baseType.columnPolicy();
-                for (String p : parts) {
-                    if (ArrayType.unnest(currentType) instanceof ObjectType objectType) {
-                        parentPolicy = objectType.columnPolicy();
-                        currentType = objectType.innerType(p);
+                if ((currentType == UndefinedType.INSTANCE) == false) {
```

Comment:
Please move this assignment in the inner scope.
```suggestion
                if ((baseType == UndefinedType.INSTANCE) == false) {
                    DataType<?> currentType = baseType;
```